File: java-version/src/main/java/bernard/tatin/threads/ThPrinterClient.java
package bernard.tatin.threads;

import bernard.tatin.threads.ThPrinter;

public class ThPrinterClient implements IThPrinterClient {
    private ThPrinter mainPrinter = ThPrinter.getMainInstance();

    public void printString(String str) {
        mainPrinter.printString(str);
    }

    public void printStrings(String[] strings) {
        mainPrinter.printStrings(strings);
    }

    public void printError(String str) {
        mainPrinter.printError(str);
    }

}

File: kotlin-version/src/main/java/bernard/tatin/Threads/ThPrinterClient.kt
package bernard.tatin.Threads

class ThPrinterClient : IThPrinterClient {
    fun printString(str: String) {
        ThPrinter.getMainInstance().printString(str)
    }

    fun printStrings(strings: Array<String>) {
        ThPrinter.getMainInstance().printStrings(strings)
    }

    fun printError(str: String) {
        ThPrinter.getMainInstance().printError(str)
    }

}
